### PR TITLE
Add args support for deno extensions

### LINKED ISSUE #416: Extensions: Allow reading CLI arguments from extensions
# Overview

There should be an API for extensions to query for the parameters passed after the extension subcommand. This way they can have their own CLI handling.

# Acceptance Criteria
- [ ] Extensions can read any CLI parameter after their subcommand
- [ ] Extensions cannot access CLI parameters before their subcommand

### PR DESCRIPTION
**Pull request by cd-work**
This makes it possible for extensions to read all parameters passed
after their subcommand using `Deno.args`.

Closes #416.

### CODE REVIEW

**Comment by cd-work** [comment]
Hoping that we can maybe get https://github.com/clap-rs/clap/pull/3885 pushed through relatively quickly.

File: cli/src/commands/extensions/extension.rs [hunk 1]
```diff
@@ -124,8 +125,18 @@ impl Extension {
 
     /// Execute an extension subcommand.
     pub async fn run(&self, api: BoxFuture<'static, Result<PhylumApi>>) -> CommandResult {
+        let args = env::args()
+            .skip_while(|arg| arg != self.name())
+            .skip(1)
+            .collect();
```
Comment (cd-work) [inline cli/src/commands/extensions/extension.rs:131]: This approach is a little unfortunate, theoretically this means that an extension with the name `phylum` will get all arguments and other similar conflicts with existing parameters. But I'm not sure we have much better options?

Comment (kylewillmon) [reply cli/src/commands/extensions/extension.rs:131]: The issue of an extension named `phylum` can be resolved with an extra `.skip(1)`. But that doesn't solve other problems...

```suggestion
        let args = env::args()
            .skip(1)
            .skip_while(|arg| arg != self.name())
            .skip(1)
            .collect();
```

Comment (cd-work) [reply cli/src/commands/extensions/extension.rs:131]: > The issue of an extension named phylum can be resolved with an extra .skip(1). But that doesn't solve other problems...

Yeah, was considering to mention that but it's really irrelevant. The problem persists either way.

**Review by kylewillmon — APPROVED** [review]

**Review by kylewillmon — APPROVED** [review]

**Pull request merged**